OpenRA.Mods.Common/Activities/HarvestResource.cs

```diff
@@ -36,6 +36,12 @@ public HarvestResource(Actor self)
 
 		public override Activity Tick(Actor self)
 		{
+			if (ChildActivity != null)
+			{
+				ChildActivity = ActivityUtils.RunActivity(self, ChildActivity);
+				return this;
```

Comment:
Is the missing `if (ChildActivity != null)` check here intentional?

Reply:
No. Fixed.